src/node_main_instance.cc

```diff
@@ -84,6 +88,17 @@ NodeMainInstance::NodeMainInstance(const SnapshotData* snapshot_data,
   SetIsolateCreateParamsForNode(isolate_params_.get());
   Isolate::Initialize(isolate_, *isolate_params_);
 
+  cpp_heap_ = CppHeap::Create(
```

Comment:
Not sure if this should be optional, or conditional on the earlier `kNoInitializeCppgc`?

Reply:
Can you add a test for `kNoInitializeCppgc`, similar to the one in `embedtest.cc`? I assume doing `CppHeap::Create()` and `Isolate:: AttachCppHeap()` right after `CommonEnvironmentSetup::Create()` would be timely enough? It would be good to know when we add something in `CommonEnvironmentSetup::Create()` that makes it no longer the case though.

I think for `NodeMainInstance()` we can probably just do this unconditionally for the add-ons. It does not seem to be necessary for embedders who use `NodeMainInstance()`  and also enable cppgc for add-ons (do they exist?) to customize this part.

Reply:
I can add `kNoInitializeCppgc` to `embedtest.cc` and have it call `cppgc::InitializeProcess` manually.  `CppHeap::Create()` and `Isolate::AttachCppHeap()` happen internally in `CommonEnvironmentSetup` however.  Are you saying you don't want that to be the case?

I think it makes sense, if Node starts allowing embedders to use `cppgc`, to expect that embedders specifying `kNoInitializeCppgc` need to manually initialize it, similar to how they are currently required to initialize v8.

Reply:
> for embedders who use `NodeMainInstance()` and also enable cppgc for add-ons (do they exist?)

I wouldn’t spend too much time worrying about embedders who don’t stick to the public API :slightly_smiling_face:

Reply:
Oh, I misread, there is only one flag for the process initialization, so it's not propagated to environments. So I guess we can just add a flag for cppgc creation to `IsolateSettingsFlags`, and then do the calls unconditionally in `CommonEnvironmentSetup::Create()` (this means that either the user cannot use `CommonEnvironmentSetup` if they want to configure the cppgc heap, but that's probably fine).